dotCMS/src/main/java/com/dotmarketing/portlets/hostvariable/ajax/HostVariableAjax.java

```diff
@@ -1,129 +1,192 @@
 import com.dotcms.repackage.org.directwebremoting.WebContextFactory;
-
 import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.NoSuchUserException;
 import com.dotmarketing.business.web.UserWebAPI;
 import com.dotmarketing.business.web.WebAPILocator;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotRuntimeException;
 import com.dotmarketing.exception.DotSecurityException;
-import com.dotmarketing.portlets.contentlet.business.HostAPI;
 import com.dotmarketing.portlets.hostvariable.bussiness.HostVariableAPI;
 import com.dotmarketing.portlets.hostvariable.model.HostVariable;
+import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.RegEX;
 import com.dotmarketing.util.UtilMethods;
 import com.liferay.portal.PortalException;
 import com.liferay.portal.SystemException;
 import com.liferay.portal.language.LanguageUtil;
 import com.liferay.portal.model.User;
 
-public class HostVariableAjax {
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
 
-	protected HostAPI hostAPI = APILocator.getHostAPI();
+/**
+ * This class allows the interaction with Site Variables in dotCMS. The methods exposed here are referenced via DWR by
+ * the UI layer to allow users to perform CRUD operations on them.
+ *
+ * @author root
+ * @since Mar 22, 2012
+ */
+public class HostVariableAjax {
 
-	public String saveHostVariable(String id, String hostId, String name, String key, String value) throws DotRuntimeException, PortalException, 
-		SystemException, DotDataException, DotSecurityException {
+	protected final UserWebAPI userWebAPI = WebAPILocator.getUserWebAPI();
+	protected final HostVariableAPI hostVariableAPI = APILocator.getHostVariableAPI();
+
+	/**
+	 * Saves a Site Variable in the specified Site.
+	 *
+	 * @param id     The ID of the Site Variable.
+	 * @param siteId The ID of the Site that the variable will be assigned to.
+	 * @param name   The human-readable name of the Site Variable.
+	 * @param key    The unique key assigned to the Site Variable.
+	 * @param value  The value of the Site Variable.
+	 *
+	 * @return If the Site Variable was saved correctly, returns {@code null}. Otherwise, an error message is returned.
+	 *
+	 * @throws DotRuntimeException  An error occurred when executing this request.
+	 * @throws PortalException      An error occurred when executing this request.
+	 * @throws SystemException      An error occurred when executing this request.
+	 * @throws DotDataException     The logged-in user does not have the required permissions to perform this action.
+	 * @throws DotSecurityException The logged-in user does not have the required permissions to perform this action.
+	 */
+	public String saveHostVariable(final String id, final String siteId, String name, String key, String value) throws DotRuntimeException, PortalException,
```

Comment:
this methos should not return a String, it should return void and throw a exception with the error message

Reply:
This is a legacy class, the method signature must remaing. Otherwise, a lot of code changes are required.